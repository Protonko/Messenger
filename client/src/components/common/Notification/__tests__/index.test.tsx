import {mount, ReactWrapper} from 'enzyme'
import {
  INotificationProps,
  Notification,
  NotificationType,
} from 'components/common/Notification'

const ELEMENT_SELECTORS = {
  success: 'notification--success',
  error: 'notification--error',
  warning: 'notification--warning',
}

describe('Notification', () => {
  let props: INotificationProps
  let component: ReactWrapper<INotificationProps>

  beforeEach(() => {
    props = {
      visible: false,
      onEntered: jest.fn(),
      text: 'Text',
    }

    component = mount(<Notification {...props} />)
  })

  it('Shouldn`t render component', () => {
    expect(component).toMatchSnapshot('Hidden notification')
  })

  it('Should render warning notification', () => {
    props.visible = true

    component = mount(<Notification {...props} />)

    expect(component.find(`.${ELEMENT_SELECTORS.warning}`).length).toBe(1)
    expect(component).toMatchSnapshot('Warning notification')
  })

  it('Should render success notification', () => {
    props.visible = true
    props.type = NotificationType.SUCCESS

    component = mount(<Notification {...props} />)

    expect(component.find(`.${ELEMENT_SELECTORS.success}`).length).toBe(1)
    expect(component).toMatchSnapshot('Success notification')
  })

  it('Should render error notification', () => {
    props.visible = true
    props.type = NotificationType.ERROR

    component = mount(<Notification {...props} />)

    expect(component.find(`.${ELEMENT_SELECTORS.error}`).length).toBe(1)
    expect(component).toMatchSnapshot('Error notification')
  })
})
