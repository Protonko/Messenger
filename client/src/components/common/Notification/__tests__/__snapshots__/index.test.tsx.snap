// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Should render error notification: Error notification 1`] = `
<Notification
  onEntered={[MockFunction]}
  text="Text"
  type="ERROR"
  visible={true}
>
  <Transition
    appear={false}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={true}
    nodeRef={
      Object {
        "current": <div
          class="notification notification--error"
        >
          <div
            class="notification__icon"
          />
          <span
            class="notification__text"
          >
            Text
          </span>
        </div>,
      }
    }
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={400}
    unmountOnExit={true}
  >
    <div
      className="notification notification--error"
    >
      <div
        className="notification__icon"
      />
      <span
        className="notification__text"
      >
        Text
      </span>
    </div>
  </Transition>
</Notification>
`;

exports[`Notification Should render success notification: Success notification 1`] = `
<Notification
  onEntered={[MockFunction]}
  text="Text"
  type="SUCCESS"
  visible={true}
>
  <Transition
    appear={false}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={true}
    nodeRef={
      Object {
        "current": <div
          class="notification notification--success"
        >
          <div
            class="notification__icon"
          />
          <span
            class="notification__text"
          >
            Text
          </span>
        </div>,
      }
    }
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={400}
    unmountOnExit={true}
  >
    <div
      className="notification notification--success"
    >
      <div
        className="notification__icon"
      />
      <span
        className="notification__text"
      >
        Text
      </span>
    </div>
  </Transition>
</Notification>
`;

exports[`Notification Should render warning notification: Warning notification 1`] = `
<Notification
  onEntered={[MockFunction]}
  text="Text"
  visible={true}
>
  <Transition
    appear={false}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={true}
    nodeRef={
      Object {
        "current": <div
          class="notification notification--warning"
        >
          <div
            class="notification__icon"
          />
          <span
            class="notification__text"
          >
            Text
          </span>
        </div>,
      }
    }
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={400}
    unmountOnExit={true}
  >
    <div
      className="notification notification--warning"
    >
      <div
        className="notification__icon"
      />
      <span
        className="notification__text"
      >
        Text
      </span>
    </div>
  </Transition>
</Notification>
`;

exports[`Notification Shouldn\`t render component: Hidden notification 1`] = `
<Notification
  onEntered={[MockFunction]}
  text="Text"
  visible={false}
>
  <Transition
    appear={false}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={true}
    nodeRef={
      Object {
        "current": null,
      }
    }
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={400}
    unmountOnExit={true}
  />
</Notification>
`;
